pkg/types/chain_writer.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/google/uuid"
+)
+
+type ChainWriter interface {
+	// SubmitTransaction packs and broadcasts a transaction to the underlying chain.
+	//
+	// The `payload` array must contain the encoded transaction payload, and the related signatures.
+	// The `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
+	SubmitTransaction(ctx context.Context, payload []any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) (int64, error)
+
+	// StatusForUUID returns the current status of a transaction in the underlying chain's TXM.
+	StatusForUUID(ctx context.Context, transactionID uuid.UUID) (TransactionStatus, error)
+
+	// GetFeeComponents retrieves the associated gas costs for executing a transaction.
+	GetFeeComponents() (ChainFeeComponents, error)
+}
+
+// TODO(nickcorin): Intentionally left blank for now.
+type TxMeta struct{}
+
+// TransactionStatus are the status we expect every TXM to support and that can be returned by StatusForUUID.
+type TransactionStatus int
+
+const (
+	Unconfirmed TransactionStatus = iota
```

Comment:
I think it makes sense to have something else before `Unconfirmed`, but specifically because to me it implies that it has been sent, so maybe something like `Unsent`? I would be OK with `Unknown` too though

Reply:
The current `TxMeta` in the `chainlink` repository is sitting in `common`.

Is there any reason to not just move it across as is?